src/scenes/DeliverablesRouter/DeliverableView.tsx

```diff
@@ -28,6 +32,8 @@ const DeliverableView = (props: Props): JSX.Element => {
   const { activeLocale } = useLocalization();
   const theme = useTheme();
 
+  const [showDocumentLimitReachedMessage, setShowDocumentLimitReachedMessage] = useState(false);
```

Comment:
We could show the message without this state.
Maybe do the calculation in this component, does not need to be triggered by the DocumentsUploader.
We can hide the DocumentsUploader in this component conditionally.

Reply:
I was thinking I would have needed to hoist the `files` state from the `DocumentUploader` up a level into the view, but the calculation needs to be performed using `deliverable.documents.length` rather than `files.length` 🤦 Good call, will update and remove this state 👍